### PR TITLE
Add interop matrix

### PR DESCRIPTION
**Pull request by t-lo**
Add initial version of the Flatcar interop matrix.

## How to use

See https://github.com/kinvolk/Flatcar/blob/t-lo/add-interop/interop-matrix.md.

### CODE REVIEW

File: interop-matrix.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Flatcar inter-operation matrix
+
+This document tracks Flatcar inter-operability across environments.
+
+Support levels:
```
Comment (vbatts) [inline interop-matrix.md]: "support" level is confusing. And does not visually convey what we're wanting here.
This mnemonic needs to go away entirely.

Instead, just have columns, so folks can visually deduce the same. Like:

| Environment | Full-Feature (release blocker) | Works | Tested | Owner | Reference (e.g. GH issue) | Notes |
|-|:-:|:-:|:-:|-|-|-|
| EC2 | | X | X | @kinvolk/flatcar-maintainers | #107 | ... |
| Azure | | X | X | @kinvolk/flatcar-maintainers |  | ... |
| GCE | X | X | X | @kinvolk/flatcar-maintainers |  |  |
| Digital Ocean (VMs) | X | X | X | @kinvolk/flatcar-maintainers |  |  |
| Equinix Metal | X | X | X | @kinvolk/flatcar-maintainers |  |  |
| ESXi / vSphere | X | X | X | @kinvolk/flatcar-maintainers |  |  |
| Hetzner Cloud | | | X | [no owner] |  |  |
| Vultr VPS  | | | X | [no owner] |  |  |

**Review by vbatts — CHANGES_REQUESTED** [review]
on the whole good, though making it very visually obvious is the big thing.

**Comment by t-lo** [comment]
Updated the PR based on the feedback provided by @vbatts .

**Review by vbatts — APPROVED** [review]
This is a good start. As folks will have links to their effort or tests, they can go here.

**Pull request merged**